src/dune_rules/mdx.ml

```diff
@@ -187,19 +187,28 @@ let syntax =
     [ ((0, 1), `Since (2, 4))
     ; ((0, 2), `Since (3, 0))
     ; ((0, 3), `Since (3, 2))
+    ; ((0, 4), `Since (3, 8))
     ]
 
-let default_files =
-  let has_extension ext s = String.equal ext (Filename.extension s) in
-  Predicate_lang.Glob.of_pred (has_extension ".md")
+let glob_predicate repr =
+  repr |> Dune_lang.Glob.of_string_exn Loc.none |> Predicate_lang.Glob.of_glob
+
+let default_files_of_version version =
+  let md_files = glob_predicate "*.md" in
+  let mld_files = glob_predicate "*.mld" in
+  let mld_support_since = (0, 4) in
+  match Syntax.Version.Infix.(version >= mld_support_since) with
+  | true -> Predicate_lang.union [ md_files; mld_files ]
+  | false -> md_files
 
 let decode =
   let open Dune_lang.Decoder in
   fields
     (let+ loc = loc
      and+ version = Dune_lang.Syntax.get_exn syntax
      and+ files =
-       field "files" Predicate_lang.Glob.decode ~default:default_files
+       let default = Predicate_lang.Standard in
```

Comment:
nitpicking but if you inline `~default:Standard` that should both save a line in the file and in the diff

Reply:
I wasn't sure whether it wouldn't be too long of a line, but it did work!